app/src/main/java/com/github/se/eventradar/viewmodel/EventsOverviewViewModel.kt

```diff
@@ -108,31 +240,34 @@ constructor(
   }
 
   fun onTabChanged(tab: Tab, state: MutableStateFlow<EventsOverviewUiState> = _uiState) {
-    state.value = state.value.copy(tab = tab)
+    // Reset search and filter when tab is changed
+    state.value =
+        state.value.copy(
+            searchQuery = "",
+            isSearchActive = false,
+            isFilterDialogOpen = false,
+            isFilterActive = false,
+            radiusQuery = "",
+            isFreeSwitchOn = false,
+            categoriesCheckedList = mutableSetOf(),
+            tab = tab)
   }
 
   fun onViewListStatusChanged(state: MutableStateFlow<EventsOverviewUiState> = _uiState) {
     state.value = state.value.copy(viewList = !state.value.viewList)
   }
-
-  fun changeFilterDialogOpen() {
-    val current = _uiState.value.isFilterDialogOpen
-    _uiState.update { currentState -> currentState.copy(isFilterDialogOpen = !current) }
-  }
-
-  fun onSearchQueryChange(newQuery: String) {
-    _uiState.update { currentState -> currentState.copy(searchQuery = newQuery) }
-  }
 }
 
 data class EventsOverviewUiState(
     val eventList: EventList = EventList(emptyList(), emptyList(), null),
     val upcomingEventList: EventList = EventList(emptyList(), emptyList(), null),
     val searchQuery: String = "",
+    val isSearchActive: Boolean = false,
     val isFilterDialogOpen: Boolean = false,
-    var radiusInputFilter: Double = -1.0,
-    var freeEventsFilter: Boolean = false,
-    val categorySelectionFilter: List<EventCategory> = emptyList(),
+    val isFilterActive: Boolean = false,
+    val radiusQuery: String = "",
+    val isFreeSwitchOn: Boolean = false,
+    val categoriesCheckedList: MutableSet<EventCategory> = mutableSetOf(),
```

Comment:
**Suggestion:** The current implementation of the category filter in the UI behaves such that no categories are selected by default, which effectively shows events from all categories. However, manually selecting all categories leads to the same result. This doesn't allow us to select no categories and could potentially confuse users about the filter's function. To improve user experience and clarity, consider initializing the application with all categories checked by default.

Reply:
Not sure when the user would want to see "no events", which is why both "no categories selected" and "all categories selected" produces the same result being "all events". I have decided to keep it as is, as per Nik's suggestion.